EIPS/eip-5851.md

```diff
@@ -0,0 +1,244 @@
+    /// @param certifying is the address that needs to be proven as the owner of the SBT defined by the tokenID.
+    /// @param `requirements` is the struct array of all the descriptions of condition metadata that is defined by the administrator. check metadata section for more information.
+    function certify(address certifying, uint256 SBFID) external returns (bool);
+
+    /// @notice function which uses the ZKProof protocol in order to validate the identity based on the given 
+    /// @dev it should only be called by the admin address.
+    /// @param SBFID is the Id of the SBT based identity certificate for which admin wants to define the Requirements.
+    /// @param certifying is the address that needs to be proven as the owner of the SBT defined by the tokenID.
+    /// @param `requirements` is the struct array of all the descriptions of condition metadata that is defined by the administrator. check metadata section for more information.
+    // eg: revoke(0xfoo,1): means that KYC admin revokes the SBT certificate number 1 for the address '0xfoo'.
+
+    function revoke(address certifying, uint256 SBFID) external returns (bool);
+```
+
+### events
+
+```solidity
+pragma solidity ^0.8.0;
+/** 
+    * standardChanged
+    * @notice standardChanged MUST be triggered when requirements are changed by the admin. 
+    * @dev standardChanged MUST also be triggered for the creation of a new SBTID.
+    e.g : emit standardChanged(1,Requirement(Metadata('depositRequirement','number', 'defines the max deposited that user can have in denomination of USDC' ), "<=", "30000");
+    defines that holder of the identifier has been changed to the condition which allows the certificate holder to call the functions with modifier , only after the deposit in the address is not greater than 30000 USDC.
+    */
+    event standardChanged(uint256 SBTID, Requirement[] _requirement);
+    /** 
+    * certified
+    * @notice certified MUST be triggered when SBT certificate is given to the certifiying address. 
+    * eg: certified( 0xfoo,2); means that wallet holder address 0xfoo is certified to hold certificate issued with id 2 , and thus can satisfy all the confitions defined by the required interface.
+    */
+    event certified(address certifying, uint256 SBTID);
+    /** 
+    * revoked
+    * @notice revoked MUST be triggered when SBT certificate is revoked. 
+    * eg : revoked( 0xfoo,1); means that entity user 0xfoo has been revoked to all the function access defined by the the SBT ID 1.
+    */
+    event revoked(address certifying, uint256 SBTID);
+```
+
+## Rationale
+
+We follow the structure of onchain metadata storage similar to that of [eip-3475](./eip-3475.md), except the fact that whole KYC requirement description is defined like the class from the eip-3475 standard but with only single condition.
+Following are the descriptions of the structures:
+
+**1.Metadata structure**:
+
+```solidity
```

Comment:
This appears to be specification. Are you sure you put this in the right section?